cdap-app-fabric/src/main/java/co/cask/cdap/app/DefaultAppConfigurer.java

```diff
@@ -217,9 +220,18 @@ public void addSchedule(Schedule schedule, SchedulableProgramType programType, S
     Preconditions.checkArgument(!programName.isEmpty(), "Program name cannot be empty.");
     Preconditions.checkArgument(!schedules.containsKey(schedule.getName()), "Schedule with the name '" +
       schedule.getName()  + "' already exists.");
-
-    ScheduleSpecification spec = new ScheduleSpecification(schedule, new ScheduleProgramInfo(programType, programName),
-                                                           properties);
+    Schedule realSchedule = schedule;
+    if (!(schedule instanceof TimeSchedule) && !(schedule instanceof StreamSizeSchedule)) {
```

Comment:
I thought this would also pass for a ``TimeSchedule`` object. But if it doesn't then great, I'll use that.